parsers/NG.py

```diff
@@ -42,19 +43,20 @@ def fetch_production(
     logger: Logger = getLogger(__name__),
 ) -> dict:
     """Requests the last known production mix (in MW) of a given zone."""
-    timestamp = (
-        arrow.get(target_datetime)
-        .to("Africa/Lagos")
-        .replace(minute=0, second=0, microsecond=0)
+
+    if target_datetime is None:
+        target_datetime = datetime.now(timezone.utc).isoformat()
+    timestamp = datetime.fromisoformat(target_datetime).replace(
+        minute=0, second=0, microsecond=0, tzinfo=TIMEZONE
     )
```

Comment:
```suggestion
    if target_datetime is None:
        target_datetime = datetime.now(timezone.utc)
    timestamp = target_datetime.replace(minute=0, second=0, microsecond=0).astimezone(
        TIMEZONE
    )
```
We don't need to convert the object to a string and then back to a datetime. Also since we are using arrows `.to()` function we should use the equivelant datetime function called `.astimezone()`.